src/main/java/io/github/classgraph/ObjectTypedValueWrapper.java

```diff
@@ -629,21 +629,21 @@ protected void toString(final boolean useSimpleNames, final StringBuilder buf) {
         } else if (stringValue != null) {
             buf.append(stringValue);
         } else if (integerValue != null) {
-            buf.append(Integer.toString(integerValue));
+            buf.append(integerValue);
```

Comment:
Is there any chance of unboxing (leading to a call to `buf.append(int)` beating out the matching of `buf.append(Object)` here?

Reply:
OK, Eclipse informs me that this does indeed apply `buf.append(Object)`. I guess I need to improve my knowledge of the semantics of boxing/unboxing and multiple dispatch!

Reply:
`integerValue` is of type `java.lang.Integer` (boxed) so `append(Object)` will be called.

Reply:
Right, I wasn't sure in what order unboxing was applied relative to the selection of method to dispatch.